import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -288,7 +288,15 @@ protected String getFormatFromBibLevel(Record record, char recordType, char bibL
                             return "Journal";
                         default: break;
                     }
+                } 
+                if (recordType == 'a' || recordType == 'm') {
+                    switch (get008Value(marc008, 23)) {
+                        case 'o':
+                            return "eJournal";
+                        default: break;
+                    }
```

Comment:
When I talked to one of our catalogers about this, he suggested that maybe the 008/21 case shouldn't take precedence over the 008/23 case in all situations. What if we changed this block to something like this?

```
            // Serial
            case 's':
                // For some materials, we may want to know if this is an online item:
                boolean isOnline = (recordType == 'a' || recordType == 'm') && (get008Value(marc008, 23) == 'o');
                // Look in 008 to determine what type of Continuing Resource
                // Make sure we have the applicable LDR/06: Language Material
                if (recordType == 'a') {
                    switch (get008Value(marc008, 21)) {
                        case 'n':
                            return "Newspaper";
                        case 'p':
                            return isOnline ? "eJournal" : "Journal";
                        default: break;
                    }
                } 
                // Default to serial even if 008 is missing (but use eJournal for online things)
                if (!isConferenceProceeding(record)) {
                    return isOnline ? "eJournal" : "Serial";
                }
                break;
```

I think that's a little more concise, and it also accounts for a bit more nuance.